Aqovia.Utilities.SagaMachine.Tests/StateMachineTests.cs

```diff
@@ -92,33 +91,23 @@ public StateMachineTests()
         [Fact(DisplayName = "Should not allow InitialiseState without Id")]
         public void SagaMachineShouldHaveInitialiseStateId()
         {
-            //Arrange
-            Guid thisId = Guid.NewGuid();
-
             _sagaMachine
                 .WithMessage<HelloMessage>((proccess, msg) => proccess
-                    .InitialiseState((hello) => new TestState
+                    .InitialiseState(hello => new TestState
                     {
                         SomeStateVariable = hello.Message
                     })
                     .Execute()
                 );
 
-            //Act
-
-            //Assert
-            Action shouldThrow = () =>
+            Func<Task> act = async () =>
             {
-
-                _sagaMachine.Handle(new HelloMessage
+                await _sagaMachine.Handle(new HelloMessage
                 {
                     Message = "Initialised"
-                }).Wait();
+                }).ConfigureAwait(false);
             };
-            shouldThrow.ShouldThrow<SagaException>();
-
-
-
+            act.ShouldThrow<SagaException>();
```

Comment:
I struggled with adding a suitable test that doesn't rely on setting maxthreads=1 in xunit.net config. (I tried using ThreadPool.SetMaxThreads to simulate but then I couldn't get it to pass!)

With regards to the Async assert, I tried re-writing the xunit.net way but found it less expressive. I'm thinking of using it as a trigger to contribute an improvement to FluentAssertions rather than writing a custom extension method!